### PR TITLE
Fix diff detection for `databricks_share` resource

### LINKED ISSUE #2167: [ISSUE] Issue with `databricks_share` update detection
### Configuration
Adding a new `object` to a `databricks_share` resource e.g.
```hcl
databricks_share my_share {
  name = "my_share"
  object {
    data_object_type = "TABLE"
    name = "main.foo.table_A"
  }
  object {
    data_object_type = "TABLE"
    name = "main.foo.table_C"
  }
}
```
->
```hcl
databricks_share my_share {
  name = "my_share"
  object {
    data_object_type = "TABLE"
    name = "main.foo.table_A"
  }
  object {
    data_object_type = "TABLE"
    name = "main.foo.table_B"
  }
  object {
    data_object_type = "TABLE"
    name = "main.foo.table_C"
  }
}
```

### Expected Behavior
Object for `main.foo.table_B` is added to the share

### Actual Behavior
Apply fails with `cannot update share: Shared Table 'foo.table_C' already exists`

When explicitly providing `shared_as` both before and after adding new object, it works.

Most likely https://github.com/databricks/terraform-provider-databricks/blob/0693e4c31f4fa84f2e01d5b0f75271f4c0b46fd8/catalog/resource_share.go#L152 does not work as expected.

### Terraform and provider versions
1.10.0

### PR DESCRIPTION
**Pull request by nkvuong**
## Changes

Improve diff detection by suppressing diff for `shared_as` field. Close #2167

## Tests

- [x] `make test` run locally
- [x] covered with integration tests in `internal/acceptance`
- [x] relevant acceptance tests are passing

### CODE REVIEW

**Comment by nkvuong** [comment]
acceptance tests passed

**Comment by nkvuong** [comment]
@marekbrysa do you mind validating this manually with your issue as well?

File: catalog/resource_share.go [hunk 1]
```diff
@@ -128,6 +128,13 @@ func (si ShareInfo) resourceShareMap() map[string]SharedDataObject {
 	return m
 }
 
+func Equal(a, b SharedDataObject) bool {
```
Comment (alexott) [inline catalog/resource_share.go:131]: I would either rename it to be more explicit on what we're comparing, or make it a member function of the corresponding struct.

**Review by alexott — COMMENTED** [review]
in general looks good, but may need to change name...

File: catalog/resource_share.go (hunk 1 above)
Comment (marekbrysa) [reply catalog/resource_share.go:131]: Also perhaps rename the parameters? `b` has the meaning of the "after"/goal state?

Comment (nfx) [reply catalog/resource_share.go:131]: not blocking, but this should be the method on `SharedDataObject` in the follow-up PR

**Review by nfx — APPROVED** [review]

**Pull request merged**